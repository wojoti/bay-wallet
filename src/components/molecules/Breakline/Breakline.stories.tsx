import {ComponentMeta, ComponentStory} from '@storybook/react-native';

import Breakline from './Breakline';

export default {
  title: 'Molecules/Breakline',
  component: Breakline,
  argTypes: {},
} as ComponentMeta<typeof Breakline>;

const Template: ComponentStory<typeof Breakline> = args => (
  <Breakline {...args} />
);

export const Example = Template.bind({});
Example.args = {
  children: 'or sign in with',
  testId: '',
};
