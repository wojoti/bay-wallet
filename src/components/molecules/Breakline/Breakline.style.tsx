import {StyleSheet} from 'react-native';
import {darkGrey, iconGrey, secondary} from 'theme/colors';

const style = StyleSheet.create({
  MaxWidthDiv: {
    width: '100%',
    marginVertical: 10,
    alignItems: 'center',
  },
  GreyLineDiv: {
    height: 1,
    width: '100%',
    backgroundColor: iconGrey,
  },
  StyledText: {
    backgroundColor: secondary,
    fontSize: 14,
    color: darkGrey,
    zIndex: 1,
    top: -13,
    paddingHorizontal: 10,
  },
});
export default style;
