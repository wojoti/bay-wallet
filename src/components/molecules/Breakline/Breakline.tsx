import {PropsWithChildren} from 'react';
import {Text, View} from 'react-native';
import style from './Breakline.style';

export type BreaklineProps = PropsWithChildren<{
  children: string;
  testId?: string;
}>;

const Breakline = ({testId, children}: BreaklineProps) => {
  return (
    <View testID={testId} style={style.MaxWidthDiv}>
      <View style={style.GreyLineDiv} />
      <Text style={style.StyledText} testID="breakline-text">
        {children}
      </Text>
    </View>
  );
};

export default Breakline;
