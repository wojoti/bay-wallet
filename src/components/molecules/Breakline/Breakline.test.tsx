import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import Breakline, {BreaklineProps} from './Breakline';

const props: BreaklineProps = {
  children: 'test breakline',
  testId: 'test-breakline-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<Breakline {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render breakline', () => {
  render(<Breakline {...props} />);
  const breaklineElement = screen.getByTestId('test-breakline-id');
  expect(breaklineElement).toBeOnTheScreen();
});

test('should render breakline with text', () => {
  render(<Breakline {...props} />);
  const breaklineElement = screen.getByTestId('test-breakline-id');
  expect(breaklineElement).toBeOnTheScreen();
  const breaklineTextElement = screen.getByTestId('breakline-text');
  expect(breaklineTextElement).toBeOnTheScreen();
  expect(breaklineTextElement).toHaveTextContent('test breakline');
});
