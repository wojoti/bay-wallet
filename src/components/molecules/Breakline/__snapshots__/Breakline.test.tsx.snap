// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "marginVertical": 10,
      "width": "100%",
    }
  }
  testID="test-breakline-id"
>
  <View
    style={
      {
        "backgroundColor": "#BDBDBD",
        "height": 1,
        "width": "100%",
      }
    }
  />
  <Text
    style={
      {
        "backgroundColor": "white",
        "color": "#595959",
        "fontSize": 14,
        "margin": 0,
        "paddingHorizontal": 10,
        "top": -13,
        "zIndex": 1,
      }
    }
    testID="breakline-text"
  >
    test breakline
  </Text>
</View>
`;
